### PR TITLE
Remove resultDependsOn/resultDependsOnSelf

### PR DESCRIPTION
**Pull request by meg-gupta**
`dependsOn` is used for lifetime dependence.

### CODE REVIEW

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/74132

@swift-ci Please smoke test

File: Sources/SwiftParser/Patterns.swift [hunk 1]
```diff
@@ -379,7 +379,7 @@ extension Parser.Lookahead {
         && !self.at(.keyword(.consuming))
         && !(experimentalFeatures.contains(.transferringArgsAndResults) && self.at(.keyword(.transferring)))
         && !(experimentalFeatures.contains(.sendingArgsAndResults) && self.at(.keyword(.sending)))
-        && !(experimentalFeatures.contains(.nonescapableTypes) && self.at(.keyword(._resultDependsOn)))
+        && !(experimentalFeatures.contains(.nonescapableTypes))
```
Comment (ahoppen) [inline Sources/SwiftParser/Patterns.swift:382]: I think you need to remove the check for experimental features as well here

**Review by atrick — APPROVED** [review]
Thanks

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/74132

@swift-ci Please smoke test

File: Sources/SwiftParser/Patterns.swift (hunk 1 above)
Comment (meg-gupta) [reply Sources/SwiftParser/Patterns.swift]: Done. Thanks for catching that

**Review by ahoppen — APPROVED** [review]

**Comment by meg-gupta** [comment]
Please test with following pull request:
https://github.com/apple/swift/pull/74132

@swift-ci Please test

**Pull request merged**